doc/sonic-build-system/SONiC-OpenSSL-FIPS-140-3.md

```diff
@@ -0,0 +1,140 @@
+# SONIC OpenSSL 140-3 HLD
+
+## Revision
+
+|  Rev  | Date       | Author     | Change Description |
+| :---: | :--------: | :--------: | ------------------ |
+|  0.1  | 2022-02-22 | Xuhui Miao | Initial version    |
+
+## Table of Contents
+- [Abbreviation](#abbreviation)
+- [Requirement](#requirement)
+- [The cryptographic modules in SONiC](#the-cryptographic-modules-in-SONiC)
+- [SONiC FIPS Configuration](#SONiC-FIPS-Configuration)
+  * [Enable FIPS on system level](#Enable-FIPS-on-system-level)
+  * [Enable FIPS on application level](#Enable-FIPS-on-application-level)
+- [OpenSSL FIPS 140-3](#OpenSSL-FIPS-140-3)
+  * [OpenSSL Engine](#OpenSSL-Engine)
+  * [SymCrypt OpenSSL Engine](#symCrypt-openSSL-engine)
+  * [OpenSSL configuration for SymCrypt Engine](#OpenSSL-configuration-for-SymCrypt-Engine)
+  * [OpenSSL configuration enhancement](#OpenSSL-configuration-enhancement)
+  * [SymCrypt OpenSSL Engine debian package](#SymCrypt-OpenSSL-Engine-debian-package)
+- [Kerberos Cryptographic Module](#Kerberos-Cryptographic-Module)
+- [Golang Cryptographic Module](#Golang-Cryptographic-Module)
+
+
+## Abbreviation
+
+| Abbreviation | Description                                  |
+| ------------ | -------------------------------------------- |
+| CAVP         | Cryptographic Algorithm Validation Program   |
+| CST          | Cryptographic and Security Test              |
+| CMVP         | Cryptographic Module Validation Program      |
+| FIPS         | Federal Information Processing Standard      |
+
+## Requirement
+SONiC only uses cryptographic modules validated by FIPS 140-3, Make SONiC compliant with FIPS 140-3.
+
+## The cryptographic modules in SONiC
+
+| Module   | Use Scenarios | Description                                                |
+| -------------------- | --------------- | -------------------------------------------- |
+| OpenSSL              | Python, OpenSSH | Cyptography and SSL/TLS ToolKit              |
+| Kerberos             | OpenSSH         | Kerboros contains builtin crypto module      |
+| Golang               | sonic-restapi   | Golang contains builtin crypt module         |
+| Libgcrypto           | GPG             | A general purpose cryptographic library originally based on code from GnuPG |
+| Kernel Crypto        | --              | Linux crypto kernel module |
```

Comment:
>-- [](http://example.com/codeflow?start=25&length=2)

Do we have use secnario in SONiC?

Reply:
No find any use scenario for SONiC, only to identify all the cryptographic modules.